symphony-bdk-http/symphony-bdk-http-api/src/main/java/com/symphony/bdk/http/api/tracing/DistributedTracingContext.java

```diff
@@ -46,6 +60,23 @@ public static void clear() {
     MDC.remove(TRACE_ID);
   }
 
+  /**
+   * Processes any custom {@link Runnable} using a given traceId. Then backup the original
+   *
+   * @param traceId a given traceId
+   * @param runnable the logic to be executed
+   */
+  public static void doWithTraceId(String traceId, Runnable runnable) {
+    final Map<String, String> backup = MDC.getCopyOfContextMap();
```

Comment:
should we back up only the trace id? here it might be misleading, if you set another MDC value part of your code, it will get removed

Reply:
Good catch, I was confusing with copying MDC context in multi-threaded context